bindings/go/evmc/host.go

```diff
@@ -182,11 +182,21 @@ func getTxContext(pCtx unsafe.Pointer) C.struct_evmc_tx_context {
 }
 
 //export getBlockHash
-func getBlockHash(pResult *C.struct_evmc_uint256be, pCtx unsafe.Pointer, number int64) {
+func getBlockHash(pResult *C.struct_evmc_uint256be, pCtx unsafe.Pointer, number int64) C.int {
 	idx := int((*C.struct_extended_context)(pCtx).index)
 	ctx := getHostContext(idx)
 
-	*pResult = evmcUint256be(ctx.GetBlockHash(number))
+	blockhash := ctx.GetBlockHash(number)
+
+	*pResult = evmcUint256be(blockhash)
+
+	r := C.int(0)
+	if blockhash != (common.Hash{}) {
+		// All zeroes hash is considered a failure in lookup.
+		// FIXME: should we instead adjust `ctx.GetBlockHash`?
```

Comment:
I think this is the main question. Leaning towards this.

Reply:
@chfast if you think that's a better idea, can you please push a new commit which changes it? (just leave the current one as it is)

Reply:
This should be changed to `blockhash, err := ctx.GetBlockHash(number)`. I can push the change later on.